iceoryx_hoofs/test/moduletests/test_shared_memory.cpp

```diff
@@ -100,11 +104,105 @@ TEST_F(SharedMemory_Test, MoveCTorWithValidValues)
+                     .failureReturnValue(SharedMemory::INVALID_HANDLE)
+                     .evaluate()
+                     .has_error());
+    auto result = iox::posix::SharedMemory::unlinkIfExist(SHM_NAME);
+    ASSERT_FALSE(result.has_error());
+    EXPECT_TRUE(*result);
 }
+
+TEST_F(SharedMemory_Test, ExclusiveCreateWorksWhenShmDoesNotExist)
+{
+    IOX_DISCARD_RESULT(iox::posix::SharedMemory::unlinkIfExist(SUT_SHM_NAME));
+    auto sut = createSharedMemory(SUT_SHM_NAME, OpenMode::EXCLUSIVE_CREATE);
+    ASSERT_FALSE(sut.has_error());
+    EXPECT_TRUE(sut->hasOwnership());
+    EXPECT_THAT(sut->getHandle(), Ne(SharedMemory::INVALID_HANDLE));
+}
+
+TEST_F(SharedMemory_Test, ExclusiveCreateFailsWhenShmExists)
+{
+    ASSERT_TRUE(createRawSharedMemory(SUT_SHM_NAME));
+    auto sut = createSharedMemory(SUT_SHM_NAME, OpenMode::EXCLUSIVE_CREATE);
+    ASSERT_TRUE(sut.has_error());
+    IOX_DISCARD_RESULT(iox::posix::SharedMemory::unlinkIfExist(SUT_SHM_NAME));
+}
+
+TEST_F(SharedMemory_Test, PurgeAndCreateWorksWhenShmDoesNotExist)
+{
+    auto sut = createSharedMemory(SUT_SHM_NAME, OpenMode::PURGE_AND_CREATE);
+    ASSERT_FALSE(sut.has_error());
+    EXPECT_TRUE(sut->hasOwnership());
+    EXPECT_THAT(sut->getHandle(), Ne(SharedMemory::INVALID_HANDLE));
+}
+
+TEST_F(SharedMemory_Test, PurgeAndCreateWorksWhenShmExists)
+{
+    ASSERT_TRUE(createRawSharedMemory(SUT_SHM_NAME));
+    auto sut = createSharedMemory(SUT_SHM_NAME, OpenMode::PURGE_AND_CREATE);
+    ASSERT_FALSE(sut.has_error());
+    EXPECT_TRUE(sut->hasOwnership());
+    EXPECT_THAT(sut->getHandle(), Ne(SharedMemory::INVALID_HANDLE));
+}
+
+TEST_F(SharedMemory_Test, CreateOrOpenCreatesShmWhenShmDoesNotExist)
+{
+    auto sut = createSharedMemory(SUT_SHM_NAME, OpenMode::OPEN_OR_CREATE);
+    ASSERT_FALSE(sut.has_error());
+    EXPECT_TRUE(sut->hasOwnership());
+    EXPECT_THAT(sut->getHandle(), Ne(SharedMemory::INVALID_HANDLE));
+}
+
+TEST_F(SharedMemory_Test, CreateOrOpenOpensShmWhenShmDoesExist)
+{
+    ASSERT_TRUE(createRawSharedMemory(SUT_SHM_NAME));
+    {
+        auto sut = createSharedMemory(SUT_SHM_NAME, OpenMode::OPEN_OR_CREATE);
+        ASSERT_FALSE(sut.has_error());
+        EXPECT_FALSE(sut->hasOwnership());
+        EXPECT_THAT(sut->getHandle(), Ne(SharedMemory::INVALID_HANDLE));
+    }
+    auto result = iox::posix::SharedMemory::unlinkIfExist(SUT_SHM_NAME);
+    ASSERT_FALSE(result.has_error());
+    EXPECT_TRUE(*result);
+}
+
+TEST_F(SharedMemory_Test, OpenWorksWhenShmExist)
+{
+    ASSERT_TRUE(createRawSharedMemory(SUT_SHM_NAME));
+    {
+        auto sut = createSharedMemory(SUT_SHM_NAME, OpenMode::OPEN_EXISTING);
+        ASSERT_FALSE(sut.has_error());
+        EXPECT_FALSE(sut->hasOwnership());
+        EXPECT_THAT(sut->getHandle(), Ne(SharedMemory::INVALID_HANDLE));
+    }
+    auto result = iox::posix::SharedMemory::unlinkIfExist(SUT_SHM_NAME);
+    ASSERT_FALSE(result.has_error());
+    EXPECT_TRUE(*result);
```

Comment:
same here

Reply:
Here I also wanted to make sure that the shm is not removed when it is not owned.